### ISSUE TITLE
over estimation of required HR for p2pool 

### ISSUE THREAD

**Issue by Cyrix126** [Cyrix126/gupaxx#9]
## OS & Version:
=< Gupaxx v1.1.1

## Bug
If the last hour average local HR is more than the sidechain HR, outside HR could be seen as negative.
When the calculation for minimum required HR on p2pool substract the negative outside HR, it will get over-estimated.

## Screenshots
mHR should not go over ~= 4.500kH/s for the difficulty at this time.
```
[2024-05-31 09:56:49.797]  Algorithm of distribution HR started for the next ten minutes.
[2024-05-31 09:56:49.797]  At least one share is in current PPLNS window.
[2024-05-31 09:56:49.797]  12.941 kH/s local HR from Xmrig
[2024-05-31 09:56:49.797]  12.925 kH/s minimum required local HR to keep a share in PPLNS window
[2024-05-31 09:56:49.797]  4.439 kH/s estimated sent the last hour for your address on p2pool, including this instance
[2024-05-31 09:56:49.797]  Hero mode is enabled for this decision
[2024-05-31 09:56:49.797]  Mining on P2pool node for 600 seconds then on XvB for 0 seconds.
```

**Comment by Cyrix126** [comment]
fixed in https://github.com/Cyrix126/gupaxx/commit/800e36345a90eabb71a9740f00fcd287376f9a7d
Issue will be closed on next release.

**Comment by Cyrix126** [comment]
closed by [1.2.0](https://github.com/Cyrix126/gupaxx/releases/tag/v1.2.0)
